lib/Sema/CSApply.cpp

```diff
@@ -2406,8 +2411,90 @@ namespace {
 
     // SWIFT_ENABLE_TENSORFLOW
     Expr *visitTFOp(ObjectLiteralExpr *expr) {
+      auto &ctx = cs.TC.Context;
+
+      auto diagnose = [&](SourceLoc loc, const Twine &message) {
+        cs.TC.diagnose(loc, diag::invalid_tfop, message.str());
+      };
+
       // All #tfop operands are RValues.
       expr->setArg(cs.coerceToRValue(expr->getArg()));
+
+      // We do most of the typechecking for #tfop here in CSApply instead of
+      // generating constraints for #tfop in CSGen because #tfop allows many
+      // different types of arguments and results that are inconvenient or
+      // impossible to generate constraints for. It's much easier to check #tfop
```

Comment:
> inconvenient or impossible to generate constraints for

can you please extend the block comment with some example(s)?

Reply:
will extend